client-tags/display-name.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "`display-name` client tag"
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    period: "2021"
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+display-name` tag name. Instead, implementations SHOULD use the
+`+draft/display-name` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification defines a client-only message tag to indicate a display name to use for the attached message, separate from the nickname or realname of the user.
+
+## Motivation
+
+This tag provides a way to provide further temporary context about the originator of a message. Example use cases include bots that relay messages from external contexts, or in role playing environments.
+
+## Architecture
+
+For the purpose of this specification, a message is defined as a `PRIVMSG`, `NOTICE`, `TAGMSG`, or [`batch`][batch] end.
+
+### Dependencies
+
+Clients wishing to use this tag MUST negotiate the [`message-tags`][tags] capability with the server.
+
+### Format
+
+Display names are indicated by a message with a `+draft/display-name` tag using the client-only prefix with the value being the desired name.
+
+## Client implementation considerations
+
+This section is non-normative.
+
+This client tag is not meant as a method of impersonating other IRC users. Take care to make it clear to users when a display name is being used. Consider formatting display names differently to nicknames and making the original nickname visible to the user.
+
+This client tag is also not meant to be used as a persistent display name for users. The [metadata framework][metadata] would be more suitable for such a use case.
+
+### Fallback string removal
+
+Clients might wish to send messages that begin with a usable fallback for other clients that don't understand the display name tag. A common existing method of indicating a display name is be to include the desired name at the start of a message, surrounded by angle brackets, or other forms of parenthesis.
+
+To support this elegantly, receiving clients that support the display name might wish to remove such a prefix if it matches the display name tag.
+
+For instance, the following message has a display name tag with the value `charles` and the message begins with the fallback string `<charles> `.
+
+```
+@+draft/display-name=charles :bot!bot@bot PRIVMSG #channel :<charles> Hello from outside IRC
+```
```

Comment:
Doesn't support `/me`; which relay bots usually encode as `* charles `.

I don't see an obvious way to provide a fallback for those.

Reply:
Thinking about this, I'm not sure it should be in scope for this tag to handle these. Relays usually send /me messages as plain PRIVMSG, so to just get rid of the prefix doesn't really do the right job. You'd have to re-encode as an ACTION CTCP to get the desired effect.

This might be the job for another tag to indicate such messages properly.